src/v/pandaproxy/schema_registry/storage.h

```diff
@@ -1004,7 +1049,8 @@ struct consume_to_store {
         // writer wins: disregard subsequent events whose seq field
         // doesn't match their actually offset.  Check is only applied
         // for messages with values, not tombstones.
-        if (val && offset != key.seq) {
+        // If no seq was provided, assume 3rdparty compatibility.
+        if (val && key.seq.has_value() && offset != key.seq) {
```

Comment:
this comment and the condition seem unaligned (but it is a bit ambiguous what is meant wrt the last part of the conditional): if `key.seq` has value then that seems to be the opposite of `no seq was provided`.

Reply:
> this comment and the condition seem unaligned (but it is a bit ambiguous what is meant wrt the last part of the conditional): if `key.seq` has value then that seems to be the opposite of `no seq was provided`.

Inside the if is the quick exit / ignore path. The work is done on the opposite condition. 3rdparty compatibility has no `seq`, so we don't fail if the `seq` doesn't match `offset`.